### PR TITLE
QF-429 Feedback email transaction azure function

### PR DESCRIPTION
**Pull request by ianatesfa**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/IApprenticeFeedbackEmailTransactionApi.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿
+using RestEase;
+using SFA.DAS.ApprenticeFeedback.Jobs.Infrastructure.Api.Requests;
+using System.Threading.Tasks;
+
+
+namespace SFA.DAS.ApprenticeCommitments.Jobs.Api
```
Comment (CodeWolf86) [inline src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/IApprenticeFeedbackEmailTransactionApi.cs]: Should be able to remove this and add in the code into the singular Outer Api Interface here: https://github.com/SkillsFundingAgency/das-apprentice-feedback-jobs/blob/main/src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/IApprenticeFeedbackApi.cs

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Startup.cs [hunk 2]
```diff
@@ -93,6 +97,11 @@ public override void Configure(IFunctionsHostBuilder builder)
                 .AddHttpMessageHandler<Http.MessageHandlers.DefaultHeadersHandler>()
                 .AddHttpMessageHandler<Http.MessageHandlers.ApimHeadersHandler>()
                 .AddHttpMessageHandler<Http.MessageHandlers.LoggingMessageHandler>();
+
+            builder.Services.AddRestEaseClient<IApprenticeFeedbackEmailTransactionApi>(url)
```
Comment (CodeWolf86) [inline src/SFA.DAS.ApprenticeFeedback.Jobs/Startup.cs]: This shouldn't be needed due to the other Outer Api already existing.

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+namespace SFA.DAS.ApprenticeFeedback.Jobs.Functions
+{
+    public class GenerateEmailTransactionsFunction
+    {
+        private readonly ILogger<GenerateEmailTransactionsFunction> _log;
+        private readonly IApprenticeFeedbackEmailTransactionApi _api;
+
+        public GenerateEmailTransactionsFunction(ILogger<GenerateEmailTransactionsFunction> log, IApprenticeFeedbackEmailTransactionApi api) //, IApprenticeFeedbackEmailTransactionService service)
```
Comment (CodeWolf86) [inline src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs]: Pass in `IApprenticeFeedbackApi` and use the Outer Api that already exists, and remove the feedback transaction service in the constructor.

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/Requests/GenerateEmailTransactionsRequest.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿using SFA.DAS.ApprenticeCommitments.Messages.Events;
+using System;
+
+namespace SFA.DAS.ApprenticeFeedback.Jobs.Infrastructure.Api.Requests
+{
+    public class GenerateEmailTransactionsRequest
```
Comment (CodeWolf86) [inline src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/Requests/GenerateEmailTransactionsRequest.cs]: Not sure you need the request if the Post just has an empty body anyway? can you get away without it?

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            _api = api;
+            _log = log;
+        }
+
+        [FunctionName("GenerateEmailTransactions")]
+        public async Task Run([TimerTrigger("%FunctionsOptions:GenerateEmailTransactionsOptions:Schedule%", RunOnStartup = false)] TimerInfo myTimer,
```
Comment (CodeWolf86) [inline src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs:27]: You may or may not want to take the logic out of the function and transfer to Nservice bus like [this](https://github.com/SkillsFundingAgency/das-apprentice-feedback-jobs/blob/main/src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/ApprenticeFeedbackSummaryTimerTrigger.cs#L17)

depending on if you think there is value in testing the event like [this](https://github.com/SkillsFundingAgency/das-apprentice-feedback-jobs/blob/main/src/SFA.DAS.ApprenticeFeedback.Jobs.UnitTests/WhenGeneratingApprenticeFeedbackSummaries.cs)

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/IApprenticeFeedbackEmailTransactionApi.cs (hunk 1 above)
Comment (ianatesfa) [reply src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/IApprenticeFeedbackEmailTransactionApi.cs]: done

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Startup.cs (hunk 2 above)
Comment (ianatesfa) [reply src/SFA.DAS.ApprenticeFeedback.Jobs/Startup.cs]: Removed

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs (hunk 3 above)
Comment (ianatesfa) [reply src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs]: done

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs (hunk 5 above)
Comment (ianatesfa) [reply src/SFA.DAS.ApprenticeFeedback.Jobs/Functions/GenerateEmailTransactionsFunction.cs:27]: Not sure what you mean by this - what logic?

File: src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/Requests/GenerateEmailTransactionsRequest.cs (hunk 4 above)
Comment (ianatesfa) [reply src/SFA.DAS.ApprenticeFeedback.Jobs/Infrastructure/Api/Requests/GenerateEmailTransactionsRequest.cs]: Removed

**Review by CodeWolf86 — APPROVED** [review]

**Pull request merged**